app/Http/Controllers/Admin/PublisherController.php

```diff
@@ -18,9 +20,13 @@ class PublisherController extends Controller
      */
     public function index()
     {
-        $publishers = Publisher::orderBy('id', 'DESC')->paginate(config('pagination.limit_page'));
+        if (Auth::user()->can('publisher.index')) {
+            $publishers = Publisher::orderBy('id', 'DESC')->paginate(config('pagination.limit_page'));
 
-        return view('admin.publisher.index', compact('publishers'));
+            return view('admin.publisher.index', compact('publishers'));
+        } else {
+            abort(Response::HTTP_FORBIDDEN);
+        }
```

Comment:
```suggestion
        abort(Response::HTTP_FORBIDDEN);
```
sửa hết những chỗ tương tự